### DISCUSSION TITLE
[V3] Components not properly re-evaluated on wire:navigate

### DISCUSSION THREAD

**Discussion by Infamoustrey** [livewire/livewire#6123] [7. Bugs (archived)] [answered]
### Livewire version

^3.0@beta

### Which PHP version are you using?

PHP 8.2

### Repository URL

https://github.com/Infamoustrey/livewire-nav-databind-bug

### Steps To Reproduce

1. Clone bug demo repo and Install deps
2. Run `php artisan serve`
3. Go to the root page and see the following 
<img width="180" alt="step-3" src="https://github.com/livewire/livewire/assets/5813665/6d23a2a9-9888-4d04-a013-0e08b19f0ea7">

4. Enter some input, the value is tracked live with no issue(uses ).
<img width="204" alt="step-4" src="https://github.com/livewire/livewire/assets/5813665/ff71ae88-76d6-4cc7-9699-950e634ed29a">

5. Then navigate to page 2 using link(uses `wire:navigate`)
<img width="164" alt="step-5" src="https://github.com/livewire/livewire/assets/5813665/6722e77d-b743-4937-b449-04a4953b1407">

6. Navigate back to page 1 using link, enter input, value is no longer live.
<img width="180" alt="step-6" src="https://github.com/livewire/livewire/assets/5813665/706f54c9-e3a1-4bff-93d7-4ccbfe48f4d5">

### Additional information

The basic components of the repo are two full page components, and a child input component that passes its value using the `Modelable` attribute. Additionally, the original input is preserved which is probably not desirable behavior if we are simulating a page load.

The relevant files are in the repository but are also listed below.

Page 1 
```php
class Page1Component extends Component
{
    public $value = '';

    public function render()
    {
        return <<<'HTML'
        <div>
            <div>
                <a href="/" wire:navigate>Page 1</a>
                <a href="/page-2" wire:navigate>Page 2</a>
            </div>
            <p>Welcome to page 1!</p>
            <p>Value: <span x-text="$wire.value"></span></p>
            <livewire:child-component wire:model="value" />
        </div>
        HTML;
    }
}
```

Page 2
```php
class Page2Component extends Component
{
    public function render()
    {
        return <<<'HTML'
        <div>
            <div>
                <a href="/" wire:navigate>Page 1</a>
                <a href="/page-2" wire:navigate>Page 2</a>
            </div>
            <p>Welcome to page 2!</p>
        </div>
        HTML;
    }
```
}

Child Component
```php 
class ChildComponent extends Component
{
    #[Modelable]
    public string $value;

    public function render()
    {
        return <<<'HTML'
        <div>
            <input type="text" wire:model="value">
        </div>
        HTML;
    }
}
```

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@Infamoustrey thanks for reporting and the thorough report! I will have a look at your repo when I get the chance 🙂

**Reply by Infamoustrey** [reply]
Appreciate it! You guys rock!

**Comment by joshhanley** [comment]
@Infamoustrey I think this is related to issue #6099. I have submitted a failing test PR #6264 covering the issue raised in #6099.

Hopefully once that is fixed, it will also fix your issue.

**Reply by Infamoustrey** [reply]
@joshhanley Yup! That looks like it's definitely the issue. Thank you so much for investigating!

**Comment by Infamoustrey** [comment]
@joshhanley looks like that got merged(woohoo), however the issue persists, i've updated the bug repo above to point at `main` which at this moment is point at [888c869](https://github.com/livewire/livewire/commit/888c869df2c886c603ffafa7b70240891dbc56ed) which includes this fix but the issue still persists

**Comment by joshhanley** [comment, marked as answer]
We're in the process of investigating all wire:navigate issues and I have tested out your scenario. I believe this has been fixed so I'm going to close this. If you're still experiencing problems, please open a new discussion. Thanks!
